modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/display/context/KBAdminNavigationDisplayContext.java

```diff
@@ -464,11 +475,83 @@ private JSONArray _getNavigationItemsJSONArray() {
 		return navigationItemsJSONArray;
 	}
 
+	private List<Long> _getSelectedItemAncestorIds() throws PortalException {
+		if (_getSelectedItemId() ==
+				KBFolderConstants.DEFAULT_PARENT_FOLDER_ID) {
+
+			return Collections.emptyList();
+		}
+
+		List<Long> selectedItemAncestorIds = new ArrayList<>();
+
+		Long kbFolderId = null;
+
+		if (_isKBArticleSelected()) {
+			KBArticle kbArticle = _getSelectedKBArticle();
+
+			if (kbArticle != null) {
+				selectedItemAncestorIds.addAll(
+					kbArticle.getAncestorResourcePrimaryKeys());
+				kbFolderId = kbArticle.getKbFolderId();
+			}
+		}
+
+		if (kbFolderId == null) {
+			kbFolderId = ParamUtil.getLong(
+				_httpServletRequest, "parentResourcePrimKey",
+				KBFolderConstants.DEFAULT_PARENT_FOLDER_ID);
+		}
+
+		KBFolder kbFolder = _getKBFolder(kbFolderId);
+
+		if (kbFolder != null) {
+			selectedItemAncestorIds.addAll(kbFolder.getAncestorKBFolderIds());
+		}
```

Comment:
About first comment: I get the folderId from the article because it is not always available from the request
About second comment: Using 0 as default value would add extra computation in this case.